source/tests/system/nirfmxwlan_driver_api_tests.cpp

```diff
@@ -339,6 +383,805 @@ TEST_F(NiRFmxWLANDriverApiTests, TXPFromExample_FetchData_DataLooksReasonable)
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_frequency_error_mean_response.frequency_error_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_symbol_clock_error_mean_response);
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_symbol_clock_error_mean_response.symbol_clock_error_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_iq_impairments_response);
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.relative_iq_origin_offset_mean());
+  EXPECT_LT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.iq_gain_imbalance_mean());
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.iq_quadrature_error_mean());
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.absolute_iq_origin_offset_mean());
+  EXPECT_LT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.iq_timing_skew_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_ppdu_type_response);
+  EXPECT_EQ(0, ofdm_mod_acc_fetch_ppdu_type_response.ppdu_type());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_mcs_index_response);
+  EXPECT_NE(0, ofdm_mod_acc_fetch_mcs_index_response.mcs_index());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_guard_interval_type_response);
+  EXPECT_EQ(0, ofdm_mod_acc_fetch_guard_interval_type_response.guard_interval_type());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_lsig_parity_check_status_response);
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_sigcrc_status_response);
+  EXPECT_EQ(-1, ofdm_mod_acc_fetch_sigcrc_status_response.sig_crc_status());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_sigbcrc_status_response);
+  EXPECT_EQ(-1, ofdm_mod_acc_fetch_sigbcrc_status_response.sigb_crc_status());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_pilot_constellation_trace_response);
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_pilot_constellation_trace_response.pilot_constellation(0).real());
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_pilot_constellation_trace_response.pilot_constellation(0).imaginary());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_data_constellation_trace_response);
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_data_constellation_trace_response.data_constellation(0).real());
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_data_constellation_trace_response.data_constellation(0).imaginary());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response);
+  EXPECT_EQ(-26, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.x0());
+  EXPECT_EQ(1, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.dx());
+  EXPECT_EQ(53, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.chain_rms_evm_per_subcarrier_mean_size());
+  EXPECT_EQ(53, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.chain_rms_evm_per_subcarrier_mean().size());
+  EXPECT_LT(0.0, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.chain_rms_evm_per_subcarrier_mean(0));
+}
+
+// Getting exception on the second internal call of BuildPortString2 (called from BuildPortString on the client):
```

Comment:
This one's interesting, it seems to be failing on the second internal call to `BuildPortString2` after we've determined the `selector_string_out_length` in `NiRFmxInstrService::BuildPortString`. I'm still working to see what the possible cause could be.

Reply:
So, for now I'm just going to add the built port strings manually and comment out the calls to BuildPortString. 

I think we should add a Task somewhere to track it but I found that if I bumped the `selector_string_out_length` in `NiRFmxInstrService::BuildPortString` by one to be `int32 selector_string_out_length = status + 1;` then the tests using BuildPortString would pass. I'm not sure if this indicates something about BuildPortString2 or if it's something on our end but we have a lot of other ivi-dance string outputs so I'd expect us to hit this earlier if it was a general issue. @astarche , thoughts?

Reply:
We actually have a size mechanism tag to deal with that: [strlen-bug](https://github.com/ni/grpc-device/search?q=strlen-bug). Although it's also possible that it's "not a bug" and the API expects us to account for the null terminator.

Also, I've noticed that `ivi-dance` typically works for the build string methods, but I'm not sure that it's documented that way. And `ivi-dance` is pretty rare in RFmx. AND I think in examples they use a fixed buffer size. 

So questions we should answer:
1. Do all of the build string APIs have the strlen bug (i.e., returns strlen for the size when strlen + 1 is needed)? Or just BuildPortString.
2. Is `ivi-dance` a cotrrect/good way to call these methods?
3. Is there a documented fixed max size that we can use instead?
4. Should we file a bug to RFmx? (doesn't affect how we resolve it, but if it's a bug we should let them know).

Reply:
Sounds good. I'll start investigating this. Added a Bug under the WLAN feature to sort this out: AB#1830349.